packages/core/test/ArrayField.test.jsx

```diff
@@ -3219,6 +3219,27 @@ describe('ArrayField', () => {
       });
     });
 
+    it('Check that when formData changes, the form should re-validate', () => {
+      const { node, rerender, compCB } = createFormComponent({
+        schema,
+        formData: [
+          {
+            text: null,
+          },
+        ],
+        liveValidate: true,
+      });
+
+      const errorMessages = node.querySelectorAll('#root_0_text__error');
+      expect(errorMessages).to.have.length(1);
+      const errorMessageContent = node.querySelector('#root_0_text__error .text-danger').textContent;
+      expect(errorMessageContent).to.contain('must be string');
+
+      rerender(compCB({ schema, formData: [{ text: 'test' }], liveValidate: true }));
```

Comment:
Since `compCB` is a React component, we should use JSX to render it, not just call it. To make it even cleaner, would it be possible to pass `Component` instead of `compCB` and then do something like this?

```suggestion
      const newProps =  { schema, formData: [{ text: 'test' }], liveValidate: true }
      rerender(<Component onSubmit={onSubmit} onError={onError} onChange={onChange} {...newProps} />);
```

Reply:
@heath-freenome 's suggestion is even better :slightly_smiling_face: